src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -605,6 +609,82 @@ public DatasetDataModel retrieveData(
     }
   }
 
+  public ColumnStatisticsModel retrieveColumnStatistics(
+      AuthenticatedUserRequest userRequest,
+      UUID datasetId,
+      String tableName,
+      String columnName,
+      String filter) {
+    Dataset dataset = retrieve(datasetId);
+
+    Column column =
+        dataset
+            .getTableByName(tableName)
+            .orElseThrow(
+                () ->
+                    new InvalidTableException(
+                        "No dataset table exists with the name: " + tableName))
+            .getColumnByName(columnName)
+            .orElseThrow(
+                () ->
+                    new InvalidColumnException(
+                        "No column exists in table "
+                            + tableName
+                            + " with column name "
+                            + columnName));
+
+    var cloudPlatformWrapper = CloudPlatformWrapper.of(dataset.getCloudPlatform());
+
+    if (cloudPlatformWrapper.isGcp()) {
+      try {
+        String bqFormattedTableName = PDAO_PREFIX + dataset.getName() + "." + tableName;
+        if (column.isDoubleType()) {
+          return BigQueryPdao.getStatsForDoubleColumn(
+              dataset, bqFormattedTableName, column, filter);
+        } else if (column.isIntType()) {
+          return BigQueryPdao.getStatsForIntColumn(dataset, bqFormattedTableName, column, filter);
+        } else if (column.isTextType()) {
+          return BigQueryPdao.getStatsForTextColumn(
+              dataset, bqFormattedTableName, tableName, column, filter);
+        }
+        return new ColumnStatisticsModel();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
```

Comment:
I don't think I've seen this handling before, is there a reason we're adding it here?

Reply:
Thanks for flagging this. I'm embarrassed to say that this is copy-pasta that I missed - I'm going to remove it.

Reply:
No embarrassment needed!